File: scripts/deploy-schema.js
#!/usr/bin/env node

/**
 * Deploy MCP Schema via Supabase REST API
 * 
 * This uses the Supabase service role to execute SQL directly
 */

const https = require('https')
const fs = require('fs')
const path = require('path')

require('dotenv').config({ path: path.join(__dirname, '..', '.env.local') })

const SUPABASE_URL = process.env.NEXT_PUBLIC_SUPABASE_URL
const SUPABASE_SERVICE_ROLE = process.env.SUPABASE_SERVICE_ROLE

if (!SUPABASE_URL || !SUPABASE_SERVICE_ROLE) {
  console.error('❌ Missing Supabase credentials')
  process.exit(1)
}

// Read SQL schema
const schemaPath = path.join(__dirname, '..', 'supabase', 'mcp_schema.sql')
const sqlContent = fs.readFileSync(schemaPath, 'utf8')

console.log('🚀 Deploying MCP Schema to Supabase')
console.log('=' .repeat(60))
console.log('')
console.log('📄 Schema file:', schemaPath)
console.log('🌐 Supabase URL:', SUPABASE_URL)
console.log('📏 SQL size:', sqlContent.length, 'bytes')
console.log('')

// Extract project ref from URL
const projectRef = new URL(SUPABASE_URL).hostname.split('.')[0]

console.log('📋 Deployment Options:')
console.log('')
console.log('Option 1: Supabase Dashboard (Easiest)')
console.log('  1. Visit: https://supabase.com/dashboard/project/' + projectRef + '/sql/new')
console.log('  2. Copy the SQL from: supabase/mcp_schema.sql')
console.log('  3. Paste into SQL Editor')
console.log('  4. Click "Run" or press Ctrl+Enter')
console.log('  5. Wait for completion (~5-10 seconds)')
console.log('')

console.log('Option 2: Using psql (If you have DB password)')
console.log('  psql "postgresql://postgres:[YOUR_PASSWORD]@db.' + projectRef + '.supabase.co:5432/postgres" -f supabase/mcp_schema.sql')
console.log('')

console.log('Option 3: Copy SQL to clipboard')
console.log('  The SQL is ready at: supabase/mcp_schema.sql')
console.log('')

console.log('📊 What will be created:')
console.log('  ✓ 6 tables (favorites, collections, items, searches, history, executions)')
console.log('  ✓ 13 indexes for performance')
console.log('  ✓ Row Level Security policies')
console.log('  ✓ 3 helper functions')
console.log('  ✓ Auto-update triggers')
console.log('')

console.log('After deployment, verify with:')
console.log('  npm run test:sync')
console.log('')

// Offer to open SQL content
const readline = require('readline')
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
})

rl.question('Show SQL content? (y/n): ', (answer) => {
  if (answer.toLowerCase() === 'y') {
    console.log('')
    console.log('=' .repeat(60))
    console.log('SQL SCHEMA CONTENT')
    console.log('=' .repeat(60))
    console.log(sqlContent)
    console.log('=' .repeat(60))
  }
  
  console.log('')
  console.log('✅ Ready to deploy!')
  console.log('')
  console.log('Recommended: Use Option 1 (Supabase Dashboard)')
  console.log('Link: https://supabase.com/dashboard/project/' + projectRef + '/sql/new')
  console.log('')
  
  rl.close()
})

File: scripts/deploy-schema.py
#!/usr/bin/env python3
"""
Deploy MCP Schema to Supabase
Uses Supabase REST API to execute SQL
"""

import os
import sys
import json
import requests
from pathlib import Path

# Load environment
env_file = Path(__file__).parent.parent / '.env.local'
env_vars = {}

if env_file.exists():
    with open(env_file) as f:
        for line in f:
            line = line.strip()
            if line and not line.startswith('#') and '=' in line:
                key, value = line.split('=', 1)
                env_vars[key] = value

SUPABASE_URL = env_vars.get('NEXT_PUBLIC_SUPABASE_URL', '')
SERVICE_ROLE = env_vars.get('SUPABASE_SERVICE_ROLE', '')

if not SUPABASE_URL or not SERVICE_ROLE:
    print('❌ Missing Supabase credentials in .env.local')
    sys.exit(1)

# Read schema
schema_file = Path(__file__).parent.parent / 'supabase' / 'mcp_schema.sql'
if not schema_file.exists():
    print(f'❌ Schema file not found: {schema_file}')
    sys.exit(1)

sql_content = schema_file.read_text()

print('🚀 Deploying MCP Schema to Supabase')
print('=' * 60)
print(f'📄 Schema: {schema_file}')
print(f'🌐 URL: {SUPABASE_URL}')
print(f'📏 Size: {len(sql_content)} bytes')
print()

# Extract project ref
project_ref = SUPABASE_URL.split('//')[1].split('.')[0]

# Try to execute via Supabase Management API
# Note: This requires Management API access which may not be available
print('📡 Attempting direct SQL execution...')
print()

# Split into statements
statements = [s.strip() for s in sql_content.split(';') if s.strip() and not s.strip().startswith('--')]

print(f'📊 Found {len(statements)} SQL statements')
print()

# For now, we'll provide instructions instead
print('⚠️  Direct SQL execution requires Database password or Management API access')
print()
print('📋 Please use one of these methods:')
print()
print('Method 1: Supabase Dashboard (Recommended)')
print(f'  Visit: https://supabase.com/dashboard/project/{project_ref}/sql/new')
print(f'  Copy SQL from: {schema_file}')
print('  Paste and click "Run"')
print()
print('Method 2: Using psql (if you have DB password)')
print(f'  psql "postgresql://postgres:[PASSWORD]@db.{project_ref}.supabase.co:5432/postgres" -f {schema_file}')
print()
print('After deployment, verify with:')
print('  npm run test:sync')
print()
